From 373c928380578195e51c2bc3e974c18bae71e155 Mon Sep 17 00:00:00 2001
From: Eric Anholt <eric@anholt.net>
Date: Thu, 7 Jan 2010 15:08:18 -0800
Patch-mainline: 2.6.34
References: fate#310916
Git-commit: a13e4093cbd3db91cc65ac8b8c2baadffa603ee9
Subject: [PATCH 0457/2588] drm/i915: Add a new mobile Sandybridge PCI ID.

Signed-off-by: Eric Anholt <eric@anholt.net>
(cherry picked from commit a13e4093cbd3db91cc65ac8b8c2baadffa603ee9)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/i915_drv.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index fd3ad01..5050cd3 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -142,6 +142,12 @@ const static struct intel_device_info intel_sandybridge_d_info = {
 	.has_hotplug = 1,
 };
 
+const static struct intel_device_info intel_sandybridge_m_info = {
+	.is_i965g = 1, .is_i9xx = 1, .need_gfx_hws = 1,
+	.has_pipe_cxsr = 1,
+	.has_hotplug = 1,
+};
+
 const static struct pci_device_id pciidlist[] = {
 	INTEL_VGA_DEVICE(0x3577, &intel_i830_info),
 	INTEL_VGA_DEVICE(0x2562, &intel_845g_info),
@@ -174,6 +180,7 @@ const static struct pci_device_id pciidlist[] = {
 	INTEL_VGA_DEVICE(0x0042, &intel_ironlake_d_info),
 	INTEL_VGA_DEVICE(0x0046, &intel_ironlake_m_info),
 	INTEL_VGA_DEVICE(0x0102, &intel_sandybridge_d_info),
+	INTEL_VGA_DEVICE(0x0106, &intel_sandybridge_m_info),
 	{0, 0, 0}
 };
 
-- 
1.7.6

